### PR TITLE
MSDKUI-1480: Guidance Dashboard overview test

### PR DESCRIPTION
**Pull request by Docik**
Guidance Dashboard overview test implementation

Signed-off-by: Aleksandr Dotsenko <1225028+Docik@users.noreply.github.com>

### CODE REVIEW

File: MSDKUIDemo/MSDKUIApp/src/androidTest/java/com/here/msdkuiapp/espresso/impl/core/CoreActions.kt [hunk 1]
```diff
@@ -68,6 +74,32 @@ open class CoreActions {
...
+                GeneralLocation.BOTTOM_CENTER,
+                OneHeightAboveCenterCoordinatesProvider,
+                Press.FINGER
+        ))
+    }
+
+    /**
+     * [CoordinatesProvider], that provides coordinates of the center of horizon line and one height above the view
+     */
+    companion object OneHeightAboveCenterCoordinatesProvider : CoordinatesProvider {
```
Comment (imvjk) [inline MSDKUIDemo/MSDKUIApp/src/androidTest/java/com/here/msdkuiapp/espresso/impl/core/CoreActions.kt]: I think u can easily do it anonymously . u can simply use 
val oneHeightAboveCenterCoordinatesProvider = CoordinatesProvider { view ->
        val locationOnScreen = IntArray(2)
        view?.getLocationOnScreen(locationOnScreen)
        val x: Float = locationOnScreen[0].plus(((view?.width?.minus(1))?.div(2.0f) ?: 0f))
        val y: Float = (locationOnScreen[1] - (view?.height ?: 0)).toFloat()
        floatArrayOf(x, y)
    }

Comment (Docik) [reply MSDKUIDemo/MSDKUIApp/src/androidTest/java/com/here/msdkuiapp/espresso/impl/core/CoreActions.kt]: Well, with inner class it looks more readable, doesn't it?

File: MSDKUIDemo/MSDKUIApp/src/androidTest/java/com/here/msdkuiapp/espresso/impl/views/drivenavigation/useractions/DriveNavigationActions.kt [hunk 2]
```diff
@@ -59,6 +65,22 @@ object DriveNavigationActions {
         return DriveNavigationMatchers
     }
 
+    /**
+     * Wait for guidance dashboard to expand
+     */
+    fun waitGuidanceDashBoardExpand(): DriveNavigationActions {
+        onRootView.perform(waitForCondition(Matchers.allOf(withText(R.string.msdkui_app_about), withParent(withParent(withId(R.id.items_list))))))
```
Comment (tstaszak89) [inline MSDKUIDemo/MSDKUIApp/src/androidTest/java/com/here/msdkuiapp/espresso/impl/views/drivenavigation/useractions/DriveNavigationActions.kt]: You can import org.hamcrest.Matchers.allOf and here just use allOf(...)

File: MSDKUIDemo/MSDKUIApp/src/androidTest/java/com/here/msdkuiapp/espresso/impl/views/drivenavigation/useractions/DriveNavigationActions.kt [hunk 3]
```diff
@@ -59,6 +65,22 @@ object DriveNavigationActions {
...
+        return this
+    }
+
+    /**
+     * Wait for guidance dashboard to collapse
+     */
+    fun waitGuidanceDashBoardCollapse(): DriveNavigationActions {
+        onRootView.perform(waitForCondition(Matchers.allOf(withText(R.string.msdkui_app_settings), withParent(withParent(withId(R.id.items_list))), not(isDisplayed()))))
```
Comment (tstaszak89) [inline MSDKUIDemo/MSDKUIApp/src/androidTest/java/com/here/msdkuiapp/espresso/impl/views/drivenavigation/useractions/DriveNavigationActions.kt]: You can import org.hamcrest.Matchers.allOf and here just use allOf(...)

File: MSDKUIDemo/MSDKUIApp/src/androidTest/java/com/here/msdkuiapp/espresso/tests/guidance/flow/GuidanceFlowTests.kt [hunk 4]
```diff
@@ -39,17 +43,20 @@ import com.here.msdkuiapp.espresso.impl.views.drivenavigation.useractions.DriveN
...
+import org.junit.FixMethodOrder
+import org.junit.After
+import org.junit.Test
 import org.junit.runners.MethodSorters
 
 /**
  * UI flow tests for guidance
  */
-@Ignore // FIXME: MSDKUI-1350
+//@Ignore // FIXME: MSDKUI-1350
```
[the remark this answers is not in the document]
Comment (Docik) [reply MSDKUIDemo/MSDKUIApp/src/androidTest/java/com/here/msdkuiapp/espresso/tests/guidance/flow/GuidanceFlowTests.kt]: ow, it should not be commented out.

File: MSDKUIDemo/MSDKUIApp/src/androidTest/java/com/here/msdkuiapp/espresso/impl/core/CoreActions.kt (hunk 1 above)
Comment (imvjk) [reply MSDKUIDemo/MSDKUIApp/src/androidTest/java/com/here/msdkuiapp/espresso/impl/core/CoreActions.kt]: actually, u can avoid creating a new class at all with the code i mentioned

File: MSDKUIDemo/MSDKUIApp/src/androidTest/java/com/here/msdkuiapp/espresso/tests/guidance/flow/GuidanceFlowTests.kt (hunk 4 above)
[the remark this answers is not in the document]
Comment (Docik) [reply MSDKUIDemo/MSDKUIApp/src/androidTest/java/com/here/msdkuiapp/espresso/tests/guidance/flow/GuidanceFlowTests.kt]: Fixed

File: MSDKUIDemo/MSDKUIApp/src/androidTest/java/com/here/msdkuiapp/espresso/impl/views/drivenavigation/useractions/DriveNavigationActions.kt (hunk 2 above)
Comment (Docik) [reply MSDKUIDemo/MSDKUIApp/src/androidTest/java/com/here/msdkuiapp/espresso/impl/views/drivenavigation/useractions/DriveNavigationActions.kt]: Fixed

File: MSDKUIDemo/MSDKUIApp/src/androidTest/java/com/here/msdkuiapp/espresso/impl/views/drivenavigation/useractions/DriveNavigationActions.kt (hunk 3 above)
Comment (Docik) [reply MSDKUIDemo/MSDKUIApp/src/androidTest/java/com/here/msdkuiapp/espresso/impl/views/drivenavigation/useractions/DriveNavigationActions.kt]: Fixed

File: MSDKUIDemo/MSDKUIApp/src/androidTest/java/com/here/msdkuiapp/espresso/impl/core/CoreActions.kt (hunk 1 above)
Comment (Docik) [reply MSDKUIDemo/MSDKUIApp/src/androidTest/java/com/here/msdkuiapp/espresso/impl/core/CoreActions.kt]: > n one more thing.. for calculting x u r using methods and for y u r using operator any reason?
not my will, compiler forced me to use methods=)

Comment (Docik) [reply MSDKUIDemo/MSDKUIApp/src/androidTest/java/com/here/msdkuiapp/espresso/impl/core/CoreActions.kt]: I've simplified this part to be easier for understanding

**Pull request merged**